### PR TITLE
Zoom to layer

### PR DESCRIPTION
**Pull request by tkohr**
This PR zooms the map preview to the selected displayed data.

TODOs:

- [x] implement logic to determine extent of data (WFS, WMS, files)
- [x] fix problem with `WmsEndpoint`
- [x] make existing tests pass

### CODE REVIEW

File: libs/feature/map/src/lib/utils/map-utils.service.ts [hunk 1]
```diff
@@ -104,4 +109,34 @@ export class MapUtilsService {
...
+      return from(
+        new WmsEndpoint(layer.url).isReady().then((endpoint) => {
+          const wmsLayer = endpoint.getLayerByName(layer.name)
+          return this.extentFromLonLat(wmsLayer.boundingBoxes['EPSG:4326'])
+        })
+      ) as Observable<Extent>
+    } else {
+      return of(undefined)
+    }
+    return of(extent)
```
Comment (jahow) [inline libs/feature/map/src/lib/utils/map-utils.service.ts:133]: This is not needed (will never be executed)

File: libs/feature/map/src/lib/utils/map-utils.service.ts [hunk 2]
```diff
@@ -104,4 +109,34 @@ export class MapUtilsService {
       return []
     }
   }
+
+  getLayerExtent(layer: MapContextLayerModel): Observable<Extent> {
+    let extent = []
+    if (layer.type === 'geojson' || layer.type === 'wfs') {
```
Comment (jahow) [inline libs/feature/map/src/lib/utils/map-utils.service.ts:115]: WFS layers do not have the data inline, they rely on a url like WMS; for the datahub we don't use WFS layers, only geosjon and WMS

File: libs/feature/map/src/lib/map-context/map-context.model.ts [hunk 3]
```diff
@@ -23,3 +24,4 @@ export interface MapContextViewModel {
   center: [number, number]
   zoom: number
 }
+export type MapContextExtentModel = [number, number, number, number]
```
Comment (jahow) [inline libs/feature/map/src/lib/map-context/map-context.model.ts:27]: I think we should document that this is expressed in the map view projection, so web mercator for now

**Review by jahow — COMMENTED** [review]
Some comments!

File: libs/feature/map/src/lib/utils/map-utils.service.ts (hunk 1 above)
Comment (tkohr) [reply libs/feature/map/src/lib/utils/map-utils.service.ts:133]: ok, shouldn't we keep a fall back for the case once we might have other types than `wms` and `geojson`?

File: libs/feature/map/src/lib/utils/map-utils.service.ts (hunk 2 above)
Comment (tkohr) [reply libs/feature/map/src/lib/utils/map-utils.service.ts:115]: ok, got it

File: libs/feature/map/src/lib/map-context/map-context.model.ts (hunk 3 above)
Comment (tkohr) [reply libs/feature/map/src/lib/map-context/map-context.model.ts:27]: yes I agree, where should I do so, here in a comment or in a readme?

Comment (jahow) [reply libs/feature/map/src/lib/map-context/map-context.model.ts:27]: a comment is enough for now I think

File: libs/feature/map/src/lib/utils/map-utils.service.ts (hunk 1 above)
Comment (jahow) [reply libs/feature/map/src/lib/utils/map-utils.service.ts]: we definitely need a fallback :) it's good now, thanks

**Review by jahow — APPROVED** [review]
Looking good, thanks a lot!

File: libs/feature/map/src/lib/map-context/map-context.model.ts (hunk 3 above)
Comment (tkohr) [reply libs/feature/map/src/lib/map-context/map-context.model.ts]: ok, I added that

**Pull request merged**